core/src/main/java/com/sequenceiq/cloudbreak/core/cluster/ClusterManagerUpgradeService.java

```diff
@@ -98,12 +99,21 @@ private void startClusterServices(Stack stack) throws CloudbreakException {
         clusterApiConnectors.getConnector(stack).startCluster();
     }
 
-    private SaltConfig createSaltConfig(Cluster cluster) {
+    private SaltConfig createSaltConfig(Long stackId, StackType stackType, Cluster cluster) {
         Map<String, SaltPillarProperties> servicePillar = new HashMap<>();
         ClouderaManagerRepo clouderaManagerRepo = clusterComponentConfigProvider.getClouderaManagerRepoDetails(cluster.getId());
-        Optional<String> license = clusterHostServiceRunner.decoratePillarWithClouderaManagerLicense(cluster.getStack().getId(), servicePillar);
+        Optional<String> license = clusterHostServiceRunner.decoratePillarWithClouderaManagerLicense(stackId, servicePillar);
         clusterHostServiceRunner.decoratePillarWithClouderaManagerRepo(clouderaManagerRepo, servicePillar, license);
         clusterHostServiceRunner.decoratePillarWithClouderaManagerSettings(servicePillar, clouderaManagerRepo);
+        decorateWorkloadClusterPillarWithCsdDownloader(stackType, cluster, servicePillar);
         return new SaltConfig(servicePillar);
     }
+
+    private void decorateWorkloadClusterPillarWithCsdDownloader(StackType stackType, Cluster cluster, Map<String, SaltPillarProperties> servicePillar) {
+        if (StackType.WORKLOAD.equals(stackType)) {
```

Comment:
Can you please cover this part with a unit test to make sure we don't update the CSDs in the case of Data Lake. If we break this then the complete Data Lake upgrade will be broken as the CSDs are not downloadable outside of Cloudera.

Reply:
Also, this could be improved that only update the CSDs if needed and those CSDs that are needed. But this does not need to be part of this PR.